p2p/starknet/handlers.go

```diff
@@ -21,15 +22,21 @@ import (
 
 type Handler struct {
 	bcReader blockchain.Reader
-	ctx      context.Context
 	log      utils.SimpleLogger
+
+	ctx    context.Context
```

Comment:
I have always read that storing a context is not a good idea. 

I can see that it is used in different protocol handlers protocol handlers when creating a stream. However, we could pass context from the `setProtocolHanderls()` and avoid this storage. The cancel function can be set when `HeadersHandler()` or any other handler function is called.

(I would be happy if this change could be made in a separate PR)

Reply:
Yeah, let me do that in separate PR

Reply:
Maybe good idea to create an issue?